giskard_vision/detectors/base.py

```diff
@@ -0,0 +1,161 @@
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Any, List, Optional, Sequence, Tuple
+
+from ..utils.errors import GiskardImportError
+
+
+@dataclass
+class ScanResult:
+    """
+    Minimum requirement for Issues to work properly
+
+    Attributes:
+        name (str): Details on transformation or slice
+        group (str): Group name for the issue
+        metric_name (str): Name of the metric
+        metric_value (float): Value of the metric on sliced or transformed dataset
+        metric_reference_value (float): Value of the metric on original dataset
+        issue_level (IssueLevel): Level of issue [Issue.MAJOR, Issue.MEDIUM, Issue.MINOR]
+        slice_size (int): Number of samples in sliced or transformed dataset
+
+    Methods:
+        get_meta_required() -> dict:
+            Returns a dictionary containing meta information required by giskard Issue
+    """
+
+    name: str
+    group: str
+    metric_name: str
+    metric_value: float
+    metric_reference_value: float
+    issue_level: str
+    slice_size: int
+    filename_examples: Optional[Sequence[str]]
+    relative_delta: float
+
+    def get_meta_required(self) -> dict:
+        # Get the meta required by the original scan API
+        deviation = f"{self.relative_delta * 100:+.2f}% than global"
+        return {
+            "metric": self.metric_name,
+            "metric_value": self.metric_value,
+            "metric_reference_value": self.metric_reference_value,
+            "deviation": deviation,
+            "slice_size": self.slice_size,
+        }
+
+
+class DetectorVisionBase:
+    """
+    Abstract class for Vision Detectors
+
+    Methods:
+        run(model: Any, dataset: Any, features: Optional[Any], issue_levels: Tuple[IssueLevel]) -> Sequence[Issues]:
+            Returns a list of giskard Issue to feed to the scan.
+
+        get_issues(model: Any, dataset: Any, results: List[ScanResult], issue_levels: Tuple[IssueLevel]):
+            Returns a list of giskard Issue from results output by get_results
+
+        get_results(model: Any, dataset: Any) -> List[ScanResult]
+            Abstract method that returns a list of ScanResult objects containing
+            evaluation results for the scan.
+    """
+
+    group: str
+    warning_messages: dict = {}
```

Comment:
this is generally dangerous. check [this](https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python).